File: app/src/java/java/com/longtoast/bilbil_api/dto/MsgEntity.java
package com.longtoast.bilbil_api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder           // ✅ builder() 메서드 생성
@NoArgsConstructor // ✅ 기본 생성자 생성 (Builder 사용 시 필수)
@AllArgsConstructor
public class MsgEntity {
    private String message;
    private Object data; // 응답 본문에 담길 실제 데이터 (KakaoDTO 등) d
}

File: app/src/main/java/com/longtoast/bilbil/dto/MsgEntity.kt
// com.longtoast.bilbil.dto/MsgEntity.kt

package com.longtoast.bilbil.dto

import com.google.gson.annotations.SerializedName

data class MsgEntity(
    val message: String,
    // 💡 핵심: data 필드를 Any (Kotlin의 일반적인 Object 타입)로 정의하여
    // Gson이 이 필드를 Map 또는 List 등 제네릭 타입으로 파싱하도록 유도합니다.
    @SerializedName("data")
    val data: Any?
)